spec/safe_statements_spec.rb

```diff
@@ -2518,6 +2518,51 @@ def up
               test_migration.suppress_messages { test_migration.migrate(:up) }
             end.to raise_error(PgHaMigrations::InvalidMigrationError, "Concurrent partitioned index creation not supported on Postgres databases before version 11")
           end
+
+          if ActiveRecord.gem_version >= Gem::Version.new("7.1") && ActiveRecord::Base.connection.postgresql_version >= 15_00_00
+            it "creates valid index with nulls_not_distinct when multiple child partitions exist" do
+              create_range_partitioned_table(:foos3, migration_klass, with_partman: true)
+
+              test_migration = Class.new(migration_klass) do
+                def up
+                  safe_add_concurrent_partitioned_index(
+                    :foos3,
+                    :text_column,
+                    nulls_not_distinct: true
+                  )
+                end
+              end
+
+              # Check that we pass the nulls_not_distinct option to the underlying add_index method
+              expect(ActiveRecord::Base.connection).to receive(:add_index).with(
+                anything,
+                anything,
+                hash_including(nulls_not_distinct: true)
+              ).at_least(:once).and_call_original
+
+              aggregate_failures do
+                expect { test_migration.suppress_messages { test_migration.migrate(:up) } }.not_to raise_error
+
+                # Verify the nulls_not_distinct property is actually set on the created indexes
```

Comment:
Small nit - I don't particularly agree this comment on top of this line of code. The line is just getting the tables with indexes that will be verified below, the actual verification isn't happening on this line.

Reply:
Yeah, I'd grant that, but that line is just setting up for the block below, so I'm inclined, if anything, to say that if personally writing it directly I'd remove the newline after this one, or even remove the temporary variable. So I'm not going to worry about it for now (there are enough other things to address with the AI changes here first).